src/plugins/vis_augmenter/public/test_helpers.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { VisLayer, VisLayerErrorTypes } from './types';
+
+export const generateVisLayer = (
+  type: any,
+  error: boolean = false,
+  resourceType: string = 'test-resource-type',
+  resourceID: string = 'test-resource-id',
+  resourceName: string = 'test-resource-name',
+  errorMessage: string = 'some-error-message'
+): VisLayer => {
+  return {
+    type,
+    originPlugin: 'test-plugin',
+    pluginResource: {
+      type: resourceType,
+      id: resourceID,
+      name: resourceName,
+      urlPath: 'test-resource-url-path',
+    },
+    error: error
+      ? {
+          type: VisLayerErrorTypes.FETCH_FAILURE,
+          message: errorMessage,
+        }
+      : undefined,
+  };
+};
```

Comment:
```suggestion
export const generateVisLayer = ({
  type,
  error,
  resource: {
    type: 'test-resource-type',
    id: 'test-resource-id',
    name: 'test-resource-name',
  }
}: {
  type: any,
  error?: string,
  resource?: {
    type: string,
    id: string,
    name: string,
  }
}): VisLayer => {
  return {
    type,
    originPlugin: 'test-plugin',
    pluginResource: {
      ...resource
      urlPath: 'test-resource-url-path',
    },
    error: error
      ? {
          type: VisLayerErrorTypes.FETCH_FAILURE,
          message: error,
        }
      : undefined,
  };
};
```

This would make the generated test layers more easier to understand.

Reply:
Thanks, makes sense to me! I'll make that change.